--- a/Rakefile
+++ b/Rakefile
@@ -3,6 +3,10 @@
 
 task :default => :test
 
+def build_dir
+  ENV['BUILDDIR'] || '.'
+end
+
 def minified_file
   ENV['FILE'] || 'mustache.min.js'
 end
@@ -26,7 +30,7 @@
 
 desc "Make a compressed build in #{minified_file}"
 task :minify => :install_uglify do
-  sh "uglifyjs mustache.js > #{minified_file}"
+  sh "uglifyjs mustache.js > #{build_dir}/#{minified_file}"
 end
 
 desc "Run JSHint"
@@ -47,10 +51,10 @@
   task short.to_sym => dependencies do
     puts "Packaging for #{name}"
 
-    mkdir_p final_location unless final_location.nil?
+    mkdir_p "#{build_dir}/#{final_location || '.'}"
 
     sources = [ "#{source}/mustache.js.pre", 'mustache.js', "#{source}/mustache.js.post" ]
-    relative_name = "#{final_location || '.'}/#{target_js}"
+    relative_name = "#{build_dir}/#{final_location || '.'}/#{target_js}"
 
     open(relative_name, 'w') do |f|
       sources.each {|source| f << File.read(source) }
